### DISCUSSION TITLE
Trigger completion provider on keystroke after initial trigger

### DISCUSSION THREAD

**Discussion by kevinslin** [microsoft/vscode-discussions#20] [Extension Development QnA] [answered]
we are writing a completion provider in vscode. by default, it gets triggered on `[`. we want to incrementally load results so that typing additional characters will re-trigger the completion (this is because we have tens of thousands of completions otherwise which introduces noticable latency). the code is like the below:

```ts
export const activate = (context: ExtensionContext) => {
  context.subscriptions.push(
    languages.registerCompletionItemProvider(
      "markdown",
      {
        provideCompletionItems,
      },
      "[", // for wikilinks and references
    )
  );
};
```

today, when you type `[`, this triggers the completion for all recent values
when the user types the next character `[f`, we would like to re-trigger the completion with a new search that is narrowed down. today we can accomplish this by manually re-triggering intellisense via keyboard shortcut but would ideally like to do this automatically as long as the user is still typing after the bracket

anyone have any suggestions on how to best accomplish this?

**Comment by isidorn** [comment, marked as answer]
Based on the discussion here https://github.com/microsoft/vscode/issues/99504
Have you set `isIncomplete` in your completion items?

**Reply by kevinslin** [reply]
yeah this was it. thanks!
